drivers/ieee802154/ieee802154_nrf5_cfg.c

```diff
@@ -0,0 +1,226 @@
+	nrf_802154_sl_ant_div_cfg_t cfg = {0};
+
+	cfg.ant_sel_pin = CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_PIN;
+	cfg.toggle_time = CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_TOGGLE_TIME;
+	cfg.ppi_ch      = CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_PPI_CHANNEL;
+	cfg.gpiote_ch   = CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_GPIOTE_CHANNEL;
+	cfg.p_timer     = NRFX_CONCAT_2(NRF_TIMER,
+		CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_TIMER_INSTANCE);
+
+	nrf_802154_antenna_diversity_config_set(&cfg);
+#endif
+
+#if CONFIG_IEEE802154_NRF5_WIFI_COEX
+	nrf_802154_wifi_coex_3wire_if_config_t coex_config =
+		NRF_802154_COEX_3WIRE_DEFAULT_CONFIG;
+
+	coex_config.request_cfg.gpio_pin =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_REQ_PIN;
+	coex_config.request_cfg.active_high =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_REQ_ACTIVE_HIGH;
+	coex_config.priority_cfg.gpio_pin =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_PRIO_PIN;
+	coex_config.priority_cfg.active_high =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_PRIO_ACTIVE_HIGH;
+	coex_config.grant_cfg.gpio_pin =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_GRA_PIN;
+	coex_config.grant_cfg.active_high =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_GRA_ACTIVE_HIGH;
+	coex_config.grant_cfg.gpiote_ch_id =
+		CONFIG_IEEE802154_NRF5_WIFI_COEX_GPIOTE_CHANNEL;
+
+	nrf_802154_wifi_coex_cfg_3wire_set(&coex_config);
+
+	nrf_802154_coex_rx_request_mode_t rx_mode;
+	nrf_802154_coex_tx_request_mode_t tx_mode;
+
+#if CONFIG_IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_ED
+	rx_mode = NRF_802154_COEX_RX_REQUEST_MODE_ENERGY_DETECTION;
+#elif CONFIG_IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_PREAMBLE
+	rx_mode = NRF_802154_COEX_RX_REQUEST_MODE_PREAMBLE;
+#elif CONFIG_IEEE802154_NRF5_WIFI_COEX_RX_REQ_MODE_DESTINED
+	rx_mode = NRF_802154_COEX_RX_REQUEST_MODE_DESTINED;
+#endif
+
+#if CONFIG_IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_FRAME_READY
+	tx_mode = NRF_802154_COEX_TX_REQUEST_MODE_FRAME_READY;
+#elif CONFIG_IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_CCA_START
+	tx_mode = NRF_802154_COEX_TX_REQUEST_MODE_CCA_START;
+#elif CONFIG_IEEE802154_NRF5_WIFI_COEX_TX_REQ_MODE_CCA_DONE
+	tx_mode = NRF_802154_COEX_TX_REQUEST_MODE_CCA_DONE;
+#endif
+
+	if (!nrf_802154_coex_rx_request_mode_set(rx_mode)) {
+		LOG_ERR("Failed to set wifi coex rx request mode");
+	}
+	if (!nrf_802154_coex_tx_request_mode_set(tx_mode)) {
+		LOG_ERR("Failed to set wifi coex tx request mode");
+	}
+
+#endif
+	return 0;
+}
+
+static int nrf5_cfg_post_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+#if !CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_MODE_DISABLED
+	nrf_802154_sl_ant_div_mode_t    rx_ant_div_mode;
+	nrf_802154_sl_ant_div_mode_t    tx_ant_div_mode;
+	nrf_802154_sl_ant_div_antenna_t ant_div_antenna;
+
+#if CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_MODE_MANUAL
+	rx_ant_div_mode = NRF_802154_SL_ANT_DIV_MODE_MANUAL;
+	tx_ant_div_mode = NRF_802154_SL_ANT_DIV_MODE_MANUAL;
+#elif CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_MODE_AUTO
+	rx_ant_div_mode = NRF_802154_SL_ANT_DIV_MODE_AUTO;
+	tx_ant_div_mode = NRF_802154_SL_ANT_DIV_MODE_MANUAL;
```

Comment:
Shouldn't the TX mode be `_AUTO` as well?

Reply:
TX does not support the `_AUTO` mode by design, so that's on purpose

Reply:
So why not initialize it outside of these `#if`s?

Reply:
Mostly to emphasize that when `_AUTO` config is selected, TX is intentionally set to `_MANUAL`. Could be moved out of `#if`s if you think it would be less ambiguous.